src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,291 @@
+        int creditBalance = workerBalanceOperations.getBalance(worker.getIdWorker());
+
+        for (GiftCodeRecord nextCode : giftCodes) {
+            if (creditBalance == 0) {
+                break;
+            }
+            if (nextCode.getAmount() <= creditBalance) {
+                payedCodes.add(nextCode);
+                creditBalance -= nextCode.getAmount();
+            }
+        }
+
+        if (creditBalance >= payOffThreshold && creditBalance >= 15) {
+            notificationText = notificationText.append("A worker has pending Payments in the amount of ").append(creditBalance).append("ct. Please add giftcodes, so the payment of the worker can be continued.").append(System.getProperty("line.separator"));
+        }
+        return payedCodes;
+    }
+
+    /**
+     * Builds the payment message and sends giftcodes to the worker.
+     *
+     * @param worker    the worker to pay
+     * @param giftCodes the giftcodes to send
+     * @throws MoneyTransferException gets thrown, if an error occured
+     */
+    private void payWorker(WorkerRecord worker, List<GiftCodeRecord> giftCodes) throws MoneyTransferException {
+        if (!giftCodes.isEmpty()) {
+            StringBuilder paymentMessage = loadMessage("src/main/resources/PaymentMessage.txt");
+            StringBuilder giftCodeMessage = new StringBuilder();
+
+            for (GiftCodeRecord rec : giftCodes) {
+                giftCodeMessage.append(rec.getCode()).append(System.getProperty("line.separator"));
+            }
+
+            //creates payment message
+            Map<String, String> map = new HashMap<>();
+            map.put("GiftCodes", giftCodeMessage.toString());
+            paymentMessage = new StringBuilder(Template.apply(paymentMessage.toString(), map));
+
+            //sends payment message and saves the giftcodes to the database
+            try {
+                mailHandler.sendMail(worker.getEmail(), "Your payment for your Crowdworking", paymentMessage.toString());
+                for (GiftCodeRecord rec : giftCodes) {
+                    workerBalanceOperations.addDebit(worker.getIdWorker(), rec.getAmount(), rec.getIdGiftCode());
+                }
+            } catch (MessagingException e) {
+                throw new MoneyTransferException("The MailHandler couldnt send mails to crowdworkers." +
+                        "It seems, that there is either a problem with the server or with the properties file.");
+            } catch (UnsupportedEncodingException e) {
+                LOGGER.error(e);
+            }
+        }
+    }
+
+    /**
+     * Sends a notification about problems or errors during payment to the administrator.
+     *
+     * @param message the message to send
+     * @param criticalError the urgency of the problem
+     * @throws MoneyTransferException gets thrown, if an error occurred
+     */
+    private void sendNotification(String message, boolean criticalError) throws MoneyTransferException {
```

Comment:
A boolean is absolutely not what we want here. We want three methods:
* `sendMail(String subject, String body)`
* `sendCriticalNotification(String message)`
* `sendNotification(String message)`

Something like that.